cmd/restic/cmd_check.go

```diff
@@ -98,6 +112,23 @@ func stringToIntSlice(param string) (split []uint, err error) {
 	return result, nil
 }
 
+// parsePercentage parses the given percentage value to a float32
+func parsePercentage(param string) (float64, error) {
```

Comment:
This comment mainly repeats the function name and the result type (which it gets wrong). It's also more complicated than it could be with the regex. How about:

```
// ParsePercentage parses a percentage string of the form "X%" where X is a float constant,
// and returns the value of that constant. It does not check the range of the value.
func parsePercentage(s string) (float64, error) {
	if !strings.HasSuffix(s, "%") {
		return 0, errors.Errorf(`parsePercentage: %q does not end in "%%"`, s)
	}
	s = s[:len(s)-1]

	p, err := strconv.ParseFloat(s, 64)
	if err != nil {
		return 0, errors.Errorf("parsePercentage: %v", err)
	}
	return p, nil
}
```

This passes all the tests except the one for -100%, but that's already covered above.

Reply:
Fixed in https://github.com/restic/restic/pull/3038/commits/4943b1ebda8ad41c5cd485e202336e780a1e8c77